cirq/contrib/qasm_import/_parser.py

```diff
@@ -35,7 +36,42 @@ def __init__(self, supported_format: bool, qelib1_include: bool,
         self.circuit = c
 
 
-class QasmParser(object):
+class QasmGate:
+
+    def __init__(self, qasm_gate: str, cirq_gate: cirq.Gate, num_args: int):
+        self.qasm_gate = qasm_gate
+        self.cirq_gate = cirq_gate
+        self.num_args = num_args
+
+    def validate_args(self, args, lineno):
+        if len(args) != self.num_args:
+            raise QasmException(
+                "{} only takes {} arg(s) (qubits and/or registers), "
+                "got: {}, at line {}".format(self.qasm_gate, self.num_args,
+                                             len(args), lineno))
+
+    def on(self, args: List[List[cirq.Qid]],
+           lineno: int = 0) -> Iterable[cirq.Operation]:
+        self.validate_args(args, lineno)
+        reg_size = 1
+        for reg in args:
+            if len(reg) > 1 and len(reg) != reg_size:
+                if reg_size == 1:
+                    reg_size = len(reg)
+                else:
+                    raise QasmException(
+                        "Non matching quantum registers of length {} "
+                        "at line {}".format([len(reg) for reg in args], lineno))
```

Comment:
This will say line 0 when the lineno argument was not specified. Maybe it should default to something less confusing like "(not specified").

Reply:
Thanks for catching this. I do not expec `lineno` to be unspecfied. I will remove the default 0 value.